src/protocol/requests.rs

```diff
@@ -98,6 +100,14 @@ pub struct TextDocumentParams {
     pub text_document: TextDocument,
 }
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TextDocumentChangeParams {
+    #[serde(rename = "textDocument")]
+    pub text_document: TextDocument,
```

Comment:
This TextDocument should be VersionedTextDocumentIdentifier,  but our `TextDocument`  is actually TextDocumentItem in https://github.com/microsoft/language-server-protocol/blob/master/versions/protocol-2-x.md#TextDocumentIdentifier

We should separate TextDocumentIdentifier, VersionedTextDocumentIdentifier, TextDocumentItem

Reply:
👍